src/aspire/source/__init__.py

```diff
@@ -115,7 +168,31 @@ def _images(self, start=0, num=None):
         :return: A 3d volume of images of size L x L x n
 
         """
-        raise NotImplementedError('Subclasses should implement this and return an Image object')
+        raise NotImplementedError('Subclasses should implement this and return a 3d ndarray')
+
+    def eval_filters(self, im_orig, start=0, num=None):
+        im = im_orig.copy()
+        end = self.n
+        if num is not None:
+            end = min(start + num, self.n)
+        all_idx = np.arange(start, end)
+
+        unique_filters = set(self.filters)
+        for f in unique_filters:
+            idx_k = np.where(self.filters[all_idx] == f)[0]
+            if len(idx_k) > 0:
+                im[:, :, idx_k] = Image(im[:, :, idx_k]).filter(f).asnumpy()
+
+        return im
+
+    def eval_filter_grid(self, L):
```

Comment:
Yes .. I added an issue #38 for this. Let's keep track of it in the Issue tracker instead of right here (will get messy to track along with everything else, and removing a method will generate several cascades of changes throughout the calling code too).